modules/openapi-generator/src/main/resources/dart/libraries/dio/serialization/built_value/class_serializer.mustache

```diff
@@ -59,14 +59,14 @@ class _${{classname}}Serializer implements PrimitiveSerializer<{{classname}}> {
     {{#anyOf}}
     {{#-first}}
     final anyOf = object.anyOf;
-    final result = {{^vendorExtensions.x-has-self-and-ancestor-only-props}}<Object?>[]{{/vendorExtensions.x-has-self-and-ancestor-only-props}}{{#vendorExtensions.x-has-self-and-ancestor-only-props}}_serializeProperties(serializers, object, specifiedType: specifiedType).toList(){{/vendorExtensions.x-has-self-and-ancestor-only-props}};
-    for (var _valueEntry in anyOf.values.entries) {
-      final _typeIndex = _valueEntry.key;
-      final _type = anyOf.types[_typeIndex];
-      final _value = _valueEntry.value;
-      result.addAll(serializers.serialize(_value, specifiedType: FullType(_type)) as Iterable<Object?>);
-    }
+    {{#vendorExtensions.x-self-and-ancestor-only-props}}
+    final result = _serializeProperties(serializers, object, specifiedType: specifiedType).toList();
+    result.addAll(serializers.serialize(anyOf, specifiedType: FullType(AnyOf, anyOf.valueTypes.map((type) => FullType(type)).toList()))!);
```

Comment:
since this infers that `serializers.serialize(anyOf)` is an Iterable, would it lead to the original problem?

Reply:
True. We need to convert the `Map` back to `Iterable` here so it can be appended to the already serialized values. Fixed.

It looks like each level of data structure embedding brings in a round of back and forth conversion. `AnyOfSerializer` returns an `Iterable`, but then `StandardJsonPlugin.afterSerialize` converts it to `Map` that we need to convert back to `Iterable` here.